lib/Driver/ToolChains/Flang.cpp

```diff
@@ -600,27 +622,42 @@ void FlangFrontend::ConstructJob(Compilation &C, const JobAction &JA,
   // Add system include arguments.
   getToolChain().AddFlangSystemIncludeArgs(Args, UpperCmdArgs);
 
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("unix");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__unix");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__unix__");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("linux");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__linux");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__linux__");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__NO_MATH_INLINES");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__LP64__");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__LONG_MAX__=9223372036854775807L");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__SIZE_TYPE__=unsigned long int");
-  UpperCmdArgs.push_back("-def"); UpperCmdArgs.push_back("__PTRDIFF_TYPE__=long int");
+  // Use clang's predefined macros
+  DiagnosticsEngine DE(new DiagnosticIDs(), new DiagnosticOptions, new IgnoringDiagConsumer());
```

Comment:
Can you use `getToolChain().getDriver().getDiags()` here instead of instantiating a new `DiagnosticsEngine`?

Reply:
`CreateTargetInfo` needs a non-const reference to a `DiagnostivsEngine` but `getToolChain().getDriver().getDiags()` gives me a const reference.

Reply:
OK.